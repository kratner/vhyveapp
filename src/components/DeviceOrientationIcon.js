import React from "react";
import { IoIosPhonePortrait } from "react-icons/io";
import styled from "styled-components";

const DeviceIconContainer = styled.div`
  position: absolute;
  top: 0.5em;
  right: 0.5em;
  z-index: 15;
  color: #fff;
  font-size: 2em;
  transition: 0.15s;
  @media only screen and (orientation: landscape) {
    top: 0.25em;
    transform: rotate(90deg);
    right: 1.5em;
  }
`;
const DeviceOrientationIcon = props => {
  const isMobileDevice = () =>
    typeof window.orientation !== "undefined" ||
    navigator.userAgent.indexOf("IEMobile") !== -1;
  return (
    <DeviceIconContainer id="deviceorientation">
      {isMobileDevice() ? <IoIosPhonePortrait /> : null}
    </DeviceIconContainer>
  );
};

export default DeviceOrientationIcon;
