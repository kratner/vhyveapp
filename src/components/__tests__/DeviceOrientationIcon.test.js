import React from "react";
import { shallow } from "enzyme";
import DeviceOrientationIcon from "../DeviceOrientationIcon";

describe("DeviceOrientationIcon", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<DeviceOrientationIcon />);
    expect(wrapper).toMatchSnapshot();
  });
});
