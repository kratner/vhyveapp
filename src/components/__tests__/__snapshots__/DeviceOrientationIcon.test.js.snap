// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceOrientationIcon renders correctly 1`] = `ShallowWrapper {}`;
